haskell-best-practices.md

```diff
@@ -13,6 +13,67 @@
      x `divEx` d = x / d
  ```
 
+### Avoid `MonadFail`
+
+Don't write our own functions using `MonadFail`/`fail`.
+
+**Bad**:
```

Comment:
This is super nitpicky, but I think this might read better if the sections were sequenced as follows:

1. Bad example
2. Good example
3. Justification